src/sentry/models/projectownership.py

```diff
@@ -175,43 +181,119 @@ def get_autoassign_owners(cls, project_id, data, limit=2):
-                *([OwnerRuleType.CODEOWNERS.value] * len(codeowners_actors)),
-            ][:limit]
-
-            # Only the first item in the list is used for assignment, the rest are just used to suggest suspect owners.
-            # So if ownership_actors is empty, it will be assigned by codeowners_actors
-            if len(ownership_actors) == 0:
-                assigned_by_codeowners = True
-
-            # The rule that would be used for auto assignment
-            auto_assignment_rule = (
-                codeowners_rules[0] if assigned_by_codeowners else ownership_rules[0]
+                return []
+
+            hydrated_ownership_rules = cls._hydrate_rules(
+                project_id, ownership_rules, OwnerRuleType.OWNERSHIP_RULE.value
+            )
+            hydrated_codeowners_rules = cls._hydrate_rules(
+                project_id, codeowners_rules, OwnerRuleType.CODEOWNERS.value
+            )
+
+            rules_in_evaluation_order = [
+                *hydrated_ownership_rules[::-1],
+                *hydrated_codeowners_rules[::-1],
+            ]
+            rules_with_owners = list(
+                filter(
+                    lambda item: len(item[1]) > 0,
+                    rules_in_evaluation_order,
+                )
             )
 
-            from sentry.models import ActorTuple
+            return rules_with_owners[:limit]
+
+    @classmethod
+    def handle_auto_assignment(cls, project_id, event):
+        """
+        Get the auto-assign owner for a project if there are any.
 
-            return (
-                ownership.auto_assignment,
-                ActorTuple.resolve_many(actors),
-                assigned_by_codeowners,
-                auto_assignment_rule,
-                actor_source,
+        We combine the schemas from IssueOwners and CodeOwners.
+
+        """
+        from sentry import analytics
+        from sentry.models import ActivityIntegration, GroupAssignee, GroupOwner, GroupOwnerType
+
+        with metrics.timer("projectownership.get_autoassign_owners"):
+            ownership = cls.get_ownership_cached(project_id)
+            queue = []
+
+            if ownership.suspect_committer_auto_assignment:
+                try:
+                    committer = GroupOwner.objects.filter(
+                        group=event.group,
+                        type=GroupOwnerType.SUSPECT_COMMIT.value,
+                        project_id=project_id,
+                    )
+                except GroupOwner.DoesNotExist:
+                    committer = []
+
+                if len(committer) > 0:
+                    queue.append(
+                        (
+                            committer[0].owner(),
+                            {
+                                "integration": ActivityIntegration.SUSPECT_COMMITTER.value,
+                            },
+                        )
+                    )
+
+            if ownership.auto_assignment:
```

Comment:
should we skip this section if we have already found an owner and details?
Seems like we only use the first one